homeassistant/components/sensor/rfxtrx.py

```diff
@@ -29,25 +31,44 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup the RFXtrx platform."""
     from RFXtrx import SensorEvent
 
+    sensors = []
+    for entity_id, entity_info in config.get('devices', {}).items():
+        if entity_id not in rfxtrx.RFX_DEVICES:
+            _LOGGER.info("Add %s rfxtrx.sensor", entity_info[ATTR_NAME])
+            event = rfxtrx.get_rfx_object(entity_info[ATTR_PACKETID])
+            new_sensor = RfxtrxSensor(event, entity_info[ATTR_NAME])
+            rfxtrx.RFX_DEVICES[entity_id] = new_sensor
+            sensors.append(new_sensor)
+
+    add_devices_callback(sensors)
+
     def sensor_update(event):
         """Callback for sensor updates from the RFXtrx gateway."""
-        if isinstance(event, SensorEvent):
-            entity_id = slugify(event.device.id_string.lower())
-
-            # Add entity if not exist and the automatic_add is True
-            if entity_id not in rfxtrx.RFX_DEVICES:
-                automatic_add = config.get('automatic_add', True)
-                if automatic_add:
-                    _LOGGER.info("Automatic add %s rfxtrx.sensor", entity_id)
-                    new_sensor = RfxtrxSensor(event)
-                    rfxtrx.RFX_DEVICES[entity_id] = new_sensor
-                    add_devices_callback([new_sensor])
-            else:
-                _LOGGER.debug(
-                    "EntityID: %s sensor_update",
+        if not isinstance(event, SensorEvent):
+            return
+
+        entity_id = "s" + slugify(event.device.id_string.lower())
```

Comment:
It's confusing that you're calling it entity_id as that's also a Home Assistant attribute but judging from the code, that is not what you mean here. Maybe call it device_id ?